### PR TITLE
Fixed spelling and grammar in most files

### PR DESCRIPTION
**Pull request by camjackson**
More proof reading for #258. Still a few more files to go.

### CODE REVIEW

File: src/index_buffer.rs [hunk 1]
```diff
@@ -2,7 +2,7 @@
 In order to draw, you need to provide a source of indices which is used to link the vertices
 together into *primitives*.
 
-There are height types of primitives, each one with a corresponding struct:
```
Comment (tomaka) [inline src/index_buffer.rs]: I can't believe I wrote that :o

File: src/uniforms/sampler.rs [hunk 2]
```diff
@@ -69,10 +69,10 @@ pub enum MinifySamplerFilter {
     /// Takes the nearest texel from the two nearest mipmaps, and merges them.
     LinearMipmapNearest,
 
-    /// Same as `Linear` but from the nearest mipmap.
+    /// Same as `Linear`, but from the nearest mipmap.
     NearestMipmapLinear,
 
-    ///
+    /// Same as `Linear`, but from the two nearest mipmaps.
```
Comment (camjackson) [inline src/uniforms/sampler.rs]: This was missing, I looked it up in the GL docs, so I'm pretty sure it's correct.

File: src/lib.rs [hunk 3]
```diff
@@ -234,7 +234,7 @@ trait GlObject {
     fn get_id(&self) -> gl::types::GLuint;
 }
 
-/// Internal trait for enums that can be turned into GLenum.
+/// Internal trait for enums that can be turned into GLEnum.
```
Comment (tomaka) [inline src/lib.rs]: That should remain `GLenum`, as it's the name of the type (see three lines below).

**Comment by tomaka** [comment]
Thanks a lot for doing that :heart_eyes:

File: src/index_buffer.rs (hunk 1 above)
Comment (camjackson) [inline src/index_buffer.rs]: Haha, guessing it was meant to say eight

File: src/lib.rs (hunk 3 above)
Comment (camjackson) [inline src/lib.rs]: Ah OK, I changed it because of the trait with capital E, although that also has a lower-case L. I'll put it back :)

**Comment by camjackson** [comment]
That's everything except for `tests`.

**Comment by tomaka** [comment]
Note that you should rebase on master instead of merging.

Can I merge this or are you adding other things to this PR?

**Comment by camjackson** [comment]
Ah right, sorry, didn't occur to me until after I'd merged it.

I'm all done now.

**Pull request merged**